notes taken during Mitch's course, Analysis-based program transformations

gasche opened the pull request:
Ping @bennn @mhyee : feel free to add your own notes.

gasche committed:
notes taken during Mitch's course, Analysis-based program transformations

tonyg commented on analysis-based program transformations/talk-notes.txt:
```diff
@@ -0,0 +1,289 @@
+
+The thesis of the course is that if you take a look at the history of
+papers you can find the deltas between one paper and the next. As
+Matthias pointed out early on, you can't trust the people as they
```
"... can't trust the *papers* as they appear out of order..."

tonyg commented on analysis-based program transformations/talk-notes.txt:
```diff
@@ -0,0 +1,289 @@
+Matthias pointed out early on, you can't trust the people as they
+appear out-of-order, and you can't trust the people as they remember
+badly. It's a really hard problem to figure out what people do and
+what they did it, that's what historians do. I want to reject the idea
```
"...and when they did it,..."

bennn commented:
ok, I'll add mine around 11pm tonight (if you decide to merge before then, I'll make a PR with the additions)

mhyee commented on analysis-based program transformations/talk-notes.txt:
```diff
@@ -0,0 +1,289 @@
+
+    ł ∈ A and ρ ⊨ ρ̂
+
+Soudness:
```
Typo: Soudness -> Sou**n**dness

mhyee commented on analysis-based program transformations/talk-notes.txt:
```diff
@@ -0,0 +1,289 @@
+wasn't exactly the same.
+
+We circulated a manuscript to them in which we pointed out that their
+copmiler were all incorrect. In their compiler, if you had
```
Typo: copmiler -> compiler

mhyee commented on analysis-based program transformations/talk-notes.txt:
```diff
@@ -0,0 +1,289 @@
+  [...]
+  use(y)
+
+they would always drop (t1) as it was useles if there was no use-of-y
```
Typo: useles -> useles**s**

mhyee commented on analysis-based program transformations/talk-notes.txt:
```diff
@@ -0,0 +1,289 @@
+messages into the future. If we don't change our message, they can't
+interact with it.
+
+tSometimes we in the field of mathematical compilers discover ideas
```
Typo: tSometimes -> Sometimes

mhyee commented:
My notes are not as detailed, so I don't have anything to add (other than pointing out typos).

bennn commented:
Committed some edits https://github.com/nuprl/hopl-s2017/pull/8/commits/66c91a5f698056afcdd3b34d903bb89ceda55683

bennn committed:
go watch arrival

- add some discussion, anecdotes, post-hoc comments
- prefix quotes with the markdown thing (`>`)

gasche commented:
Do I correctly understand that this is now good to merge? Anything still to be done?

bennn commented:
merged
https://github.com/nuprl/hopl-s2017/commit/c901ceb165300bf5dce32135446cf586077746ae
https://github.com/nuprl/hopl-s2017/commit/03b922df86aa547f86c991545729c1dffca9a48a
